crates/jxl-grid/src/test/subgrids.rs

```diff
@@ -86,3 +89,193 @@ fn mutable_subgrid_split_merge() {
+}
+
+#[test]
+fn mutable_subgrid_swap(){
+    let mut buf = [1, 2, 3, 4];
+    let mut sub = MutableSubgrid::<u32>::from_buf(&mut buf, 2, 2, 2);
+    sub.swap((0, 0), (1, 1));
+    assert_eq!(buf, [4, 2, 3, 1]);
+}
+
+#[test]
+fn mutable_subgrid_borrow_mut(){
+    let mut buf = [1, 2, 3, 4];
+    let mut grid = MutableSubgrid::from_buf(&mut buf, 4, 1, 4);
+    let m2 = grid.borrow_mut();
+    assert_eq!(m2.width(), 4);
+    assert_eq!(m2.height(), 1);
+}
+
+#[test]
+fn mutable_subgrid_as_shared(){
+    let mut buf = [1, 2, 3, 4];
+    let grid = MutableSubgrid::from_buf(&mut buf, 4, 1, 4);
+    let s2 = grid.as_shared();
+    assert_eq!(s2.width(), 4);
+    assert_eq!(s2.height(), 1);
+}
+
+#[test]
+#[should_panic]
+fn mutable_subgrid_split_horizontal_with_index_exceeds_width(){
+    let mut buf = [0, 1, 2];
+    let mut grid = MutableSubgrid::from_buf(&mut buf, 3, 1, 3);
+    let _ = grid.split_horizontal(4);
+}
+
+#[test]
+#[should_panic]
+fn mutable_subgrid_split_split_horizontal_in_place_with_index_exceeds_width(){
+    let mut buf = [0, 1, 2];
+    let mut grid = MutableSubgrid::from_buf(&mut buf, 3, 1, 3);
+    let _ = grid.split_horizontal_in_place(4);
+}
+
+#[test]
+#[should_panic]
+fn mutable_subgrid_split_split_vertical_with_index_exceeds_height(){
+    let mut buf = [0];
+    let mut grid = MutableSubgrid::from_buf(&mut buf, 1, 1, 1);
+    let _ = grid.split_vertical(2);
+}
+
+#[test]
+#[should_panic]
+fn mutable_subgrid_split_split_vertical_in_place_with_index_exceeds_height(){
+    let mut buf = [0];
+    let mut grid = MutableSubgrid::from_buf(&mut buf, 1, 1, 1);
+    let _ = grid.split_vertical_in_place(2);
+}
+
+#[test]
+fn mutable_subgrid_into_groups(){
+    let mut buf = [1, 2, 3, 4];
+    let grid1 = MutableSubgrid::from_buf(&mut buf, 2, 2, 2);
+    let group1 = grid1.into_groups(1, 1);
+    assert_eq!(group1.len(), 4);
+    for g in group1 {
+        assert_eq!(g.width(), 1);
+        assert_eq!(g.height(), 1);
+    }
+    let grid2 = MutableSubgrid::from_buf(&mut buf, 2, 2, 2);
+    let group2 = grid2.into_groups(3, 3);
+    assert_eq!(group2.len(), 1);
+    let g = &group2[0];
+    assert_eq!(g.width(), 2);
+    assert_eq!(g.height(), 2);
+}
+
+#[test]
+fn mutable_subgrid_into_groups_with_fix_count(){
+    let mut buf = [0, 1, 2, 3, 4, 5, 6, 7, 8];
+    let grid = MutableSubgrid::from_buf(&mut buf, 3, 3, 3);
+    let groups = grid.into_groups_with_fixed_count(2, 2, 2, 2);
+    assert_eq!(groups.len(), 4);
+    assert_eq!(groups[0].width(), 2);
+    assert_eq!(groups[0].height(), 2);
+    assert_eq!(groups[1].width(), 1);
+    assert_eq!(groups[1].height(), 2);
+    assert_eq!(groups[2].width(), 2);
+    assert_eq!(groups[2].height(), 1);
+    assert_eq!(groups[3].width(), 1);
+    assert_eq!(groups[3].height(), 1);
+}
+
+#[test]
+fn mutable_subgrid_as_vectored(){
+    let mut data = vec![1.0; 8];
+    let mut msg = MutableSubgrid::from_buf(&mut data[..], 4, 2, 4);
+    let opt = msg.as_vectored::<float32x4_t>();
```

Comment:
This is only available in aarch64, so the test should be gated behind `#[cfg(all(target_arch = "aarch64", target_feature = "neon"))]`.

Reply:
Well, having `target_feature = "neon"` in compile time is a bit too much, maybe using `is_aarch64_feature_detected!` is better...